convention.md

```diff
@@ -18,11 +18,49 @@ The underscore (`_`) is used and reserved for Homie *node arrays*.
 
 ### Payload
 
-Every MQTT message payload MUST be sent as UTF-8 encoded string.
-A numeric data type MUST be converted to a literal string with the dot character as the decimal separator for the `float` data type.
-An empty message is not valid as payload for numeric data types.
-The value published as payload MUST be valid for the respective property/attribute type as per the convention definition below.
-Booleans MUST be converted to "true" or "false".
+ - Every MQTT message payload MUST be sent as a UTF-8 encoded string
+-  The value published as payload MUST be valid for the respective property/attribute type as per the defined convention
+ 
+String
+ - String types are limited to 268,435,456 characters 
+ - An empty string ("") is a valid payload
+ 
+Integer
+ - Integer types are UTF-8 encoded representations of 64-bit signed whole numbers
+ - Integers range from -9,223,372,036,854,775,808 (-2<sup>63</sup>) to 9,223,372,036,854,775,807 (2<sup>63</sup>-1)
+ - The payload may only contain whole numbers and the negation character "-". No other characters including spaces (" ") are permitted 
+ - A string with just a negation sign ("-") is not a valid payload
+ - An empty string ("") is not a valid payload
+ 
+Float
+ - Float types are UTF-8 encoded representations of 64-bit signed floating point numbers
+ - Floats range from 2<sup>-1074</sup> to (2-2<sup>-52</sup>)*2<sup>1023</sup>
+ - The payload may only contain whole numbers, the negation character "-", and the decimal separator ".", no other characters, including spaces (" ") are permitted 
+ - The dot character (".") is the decimal separator (used if necessary) and may only have a single instance present in the payload
+ - Representations of numeric concepts such as "NaN" (Not a Number) and "Infinity" are not a valid payload
```

Comment:
I agree but I wonder if we should shortly justify this decision

Reply:
My assumption is that they are not broadly supported.

- Is that assumption valid?
- Is that enough to justify exclusion?

Reply:
I'm not too certain about that. NaN and Inf are supported by many implementations (even though always slightly different). My reasoning comes ones again from the convention: A property represents a real *property* of a device, e.g. a temperature. The device should always be able to publish a valid value or none at all. The case of a property with non-continuous state (i.e. without continuous update) was discussed in other issues and relies on either the stateless concept, another property (e.g.` TempMeasurementActive`), or a user-defined logic (not preferred but of course an option).

Do others agree?

Reply:
My inclination is to not justify the exclusion of NaN or Infinity.  I am not sure how to concisely articulate the reasoning and I think people may use homie for things that don't strictly adhere to the definition above of a "real property".

Do you have a suggestions for a concise justification?